ee_tests/src/spec_strategy/deployment_spec_strategy.ts

```diff
@@ -0,0 +1,131 @@
+// tslint:disable:max-line-length
+import { browser, element, by, ExpectedConditions as until, $, $$ } from 'protractor';
+import * as support from '../support';
+import { SpaceDeploymentsPage, DeploymentStatus, DeployedApplication, DeployedApplicationEnvironment, Environment, ResourceUsageData } from '../page_objects/space_deployments.page';
+import { SpacePipelinePage } from '../page_objects/space_pipeline.page';
+
+export abstract class DeploymentStrategy {
+
+    protected spaceName: string;
+
+    protected spacePipelinePage: SpacePipelinePage;
+
+    protected spaceDeploymentsPage: SpaceDeploymentsPage;
+
+    constructor(spaceName: string, spacePipelinePage: SpacePipelinePage) {
+        this.spaceName = spaceName;
+        this.spacePipelinePage = spacePipelinePage;
+        this.spaceDeploymentsPage = new SpaceDeploymentsPage();
+    }
+
+    public async showDeploymentsScreen() {
+        support.info('Verifying deployments page');
+        await this.spacePipelinePage.spaceHeader.deploymentsOption.clickWhenReady();
+        await browser.sleep(5000);
+        this.spaceDeploymentsPage = new SpaceDeploymentsPage();
+    }
+
+    public async verifyApplication(): Promise<DeployedApplication> {
+        support.info('Verifying deployed applications');
+        let applications = await this.spaceDeploymentsPage.getDeployedApplications();
+        expect(applications.length).toBe(1, 'number of deployed applications');
+
+        let application = applications[0];
+        expect(application.getName()).toBe(this.spaceName, 'application name');
```

Comment:
Not necessary in this PR but shouldn't we make this class more general, i.e. no hard-coded values but rather pass these expected values to the function `verifyApplication(1, spaceName)`